test/gtest/ucp/test_ucp_request.cc

```diff
@@ -100,3 +102,398 @@ UCS_TEST_P(test_ucp_request, test_request_query)
 }
 
 UCP_INSTANTIATE_TEST_CASE_TLS(test_ucp_request, all, "all")
+
+class test_proto_reset : public ucp_test {
+public:
+    enum operation_e {
+        TAG,
+        RMA_GET,
+        RMA_PUT,
+        STREAM,
+        AM
+    };
+
+    test_proto_reset() : m_completed(false), m_am_cb_cnt(0)
+    {
+    }
+
+    void init() override
+    {
+        if (!m_ucp_config->ctx.proto_enable) {
+            UCS_TEST_SKIP_R("reset is not supported for proto v1");
+        }
+
+        if (is_self()) {
+            UCS_TEST_SKIP_R("self transport has no pending queue");
+        }
+
+        ucp_test::init();
+        modify_config("TCP_SNDBUF", "8K", IGNORE_IF_NOT_EXIST);
+        modify_config("IB_TX_QUEUE_LEN", "65", IGNORE_IF_NOT_EXIST);
+        create_entity(true);
+
+        sender().connect(&receiver(), get_ep_params());
+        receiver().connect(&sender(), get_ep_params());
+    }
+
+    void cleanup() override
+    {
+        m_rkeys.clear();
```

Comment:
What about `m_sbufs`? That's a vector of vectors..

Reply:
will be released in dtor